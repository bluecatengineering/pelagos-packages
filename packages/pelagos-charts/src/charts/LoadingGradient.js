import PropTypes from 'prop-types';

const list = [
	[0, 'stop-bg'],
	[0.125, 'stop-fg'],
	[0.25, 'stop-bg'],
];

const LoadingGradient = ({id, className}) => (
	<defs>
		<linearGradient id={id} className={className} gradientUnits="userSpaceOnUse">
			{list.map(([offset, className]) => (
				<stop key={offset} className={className} offset={offset}>
					<animate
						attributeName="offset"
						dur="3s"
						repeatCount="indefinite"
						from={offset}
						to={offset + 0.75}
						calcMode="spline"
						keyTimes="0;1"
						keySplines=".5 0 .5 1"
					/>
				</stop>
			))}
		</linearGradient>
	</defs>
);

LoadingGradient.propTypes = {
	id: PropTypes.string,
	className: PropTypes.string,
};

export default LoadingGradient;
