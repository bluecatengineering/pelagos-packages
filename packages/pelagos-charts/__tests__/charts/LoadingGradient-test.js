import {shallow} from 'enzyme';

import LoadingGradient from '../../src/charts/LoadingGradient';

jest.unmock('../../src/charts/LoadingGradient');

describe('LoadingGradient', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<LoadingGradient id="test" className="TestClass" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
