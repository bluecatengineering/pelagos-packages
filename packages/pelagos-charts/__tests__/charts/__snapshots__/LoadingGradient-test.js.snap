// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingGradient rendering renders expected elements 1`] = `
<defs>
  <linearGradient
    className="TestClass"
    gradientUnits="userSpaceOnUse"
    id="test"
  >
    <stop
      className="stop-bg"
      offset={0}
    >
      <animate
        attributeName="offset"
        calcMode="spline"
        dur="3s"
        from={0}
        keySplines=".5 0 .5 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        to={0.75}
      />
    </stop>
    <stop
      className="stop-fg"
      offset={0.125}
    >
      <animate
        attributeName="offset"
        calcMode="spline"
        dur="3s"
        from={0.125}
        keySplines=".5 0 .5 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        to={0.875}
      />
    </stop>
    <stop
      className="stop-bg"
      offset={0.25}
    >
      <animate
        attributeName="offset"
        calcMode="spline"
        dur="3s"
        from={0.25}
        keySplines=".5 0 .5 1"
        keyTimes="0;1"
        repeatCount="indefinite"
        to={1}
      />
    </stop>
  </linearGradient>
</defs>
`;
